src/main/java/edu/kis/vh/nursery/stack/IntArrayStack.java

```diff
@@ -40,7 +40,7 @@ public boolean callCheck() {
      */
     @Override
     public boolean isFull() {
-        return total == STACK_CAPACITY - 1;
+        return total == STACK_CAPACITY;

```

Comment:
Changing IS_EMPTY value requires such fixes? It does not make sense. Incorrect fix.
There is no reasonable relation between the default returned value and isFull logic!
Simply: total variable initialization and validation (callCheck) should ALWAYS remain -1.